internal/terraform/terraform.go

```diff
@@ -333,3 +289,71 @@ func OutputSingleValue(dryRun bool, directory, tfEntrypoint, outputName string)
 
 	log.Println("tfOutput is: ", tfOutput.String())
 }
+
+// ApplyUsersTerraform load environment variables into the host based on the git provider, change directory to the
+// Terraform required modules, terraform init, terraform apply and clean terraform files.
+// todo: break it into smaller functions with no dependencies in order to allow unit tests
+func ApplyUsersTerraform(dryRun bool, directory string, gitProvider string) error {
+
+	config := configs.ReadConfig()
+
+	if viper.GetBool("create.terraformapplied.users") || dryRun {
+		log.Println("skipping: ApplyUsersTerraform")
+		return nil
+	}
+
+	if len(gitProvider) == 0 {
+		return errors.New("git provider not provided, skipping terraform apply")
+	}
+
```

Comment:
I am reflecting if this is an event that we should break the installation. We shouldn't have reached this far on an installation without a `gitProvider`. That is why we were using the `bool` flavor to avoid this check. 

I double-checked, it seems before a flavor creating we now set the `git.mode` preventing this to be empty: https://github.com/kubefirst/kubefirst/blob/main/cmd/create.go#L78